odoo/tools/split_table.py

```diff
@@ -0,0 +1,41 @@
+from itertools import islice
+from lxml import etree
+
+def copy_attributes(src, dst, exclude=()):
+    """
+    Copies all attributes from src to dst. If exclude is not None, the
+    attributes in exclude are not copied.
+
+    :param src: The source element
+    :param dst: The destination element
+    :param exclude: A list of attributes that should not be copied
+    """
+    for key, value in src.attrib.items():
+        if key not in exclude:
+            dst.set(key, value)
+
+def split_table(tree, max_rows):
+    """
+    Walks through the etree and splits tables with more than max_rows rows into
+    multiple tables with max_rows rows.
+
+    This function is needed because wkhtmltopdf has a exponential memory growth
+    when processing tables with many rows. This function is a workaround for
+    this problem.
+
+    :param tree: The etree to process
+    :param max_rows: The maximum number of rows per table
+    """
+    tables_to_split = [t for t in tree.iter('table') if len(t) > max_rows]
+    for table in tables_to_split:
+        prev = table
+        nrows = len(table)
+        while nrows > max_rows:
```

Comment:
```suggestion
        for _ in range((len(table) - 1) // max_rows)
```
Then you don't need `nrows` and `nrows -= max_rows`